docs/guides/app-payments/build-on-mobile.mdx

```diff
@@ -92,11 +247,81 @@ void openMobilePay(String deepLink) {
     }
 }
 ```
-
 </TabItem>
+</Tabs>
+
+After this, the MobilePay app will be launched and the user will be able to proceed with the payment flow.
+
+## Returning to Your App
+
+There are multiple ways a user can return to your app:
+
+- User approves the payment and is redirected back to your app.
+- User cancels the payment and is redirected back to your app.
+- User fails to complete the payment due to some kind of error and is redirected back to your app.
+- User closes the app before/after reservation using the native app switcher.
+- User presses back in the MobilePay Login screen.
+
+All these can be grouped into:
+
+- App switching.
+- Redirect.
+
+The following section addresses how each one should be handled.
+
+### App Switching
+
+At any point user can close MobilePay app and just return to your app. **This can happen even if payment is already reserved!** Since it is unknown if your app was opened after completed payment, you'll need to check payment state each time your app returns to the foreground. For this, you should use your backend service that provides current state of payment (more details [here](/docs/payments-refunds/create-payments)).
+
+### Redirect from MobilePay
+
+The other scenario is that the MobilePay app tries to automatically navigate back to your app. For this to work you'll need:
+
+- Valid deep link to a place that knows how to proceed with the flow ([official deep linking guide](https://developer.android.com/training/app-links/deep-linking)).
+- Provide this deep link during [payment creation](#create-payment).
+
+In the general case, this deep link should open the app screen which handles the "App switching" scenario described above, and just treat it the same way as if the user manually navigated to your app.
+
+:::tip Payment state in your deep link
+
+When redirecting back to your app, MobilePay appends a query parameter `status` to the end of a deep link. It can contain these values:
+
+- successful
+- failed
+- rejected
+
+Consider checking this parameter and triggering a capture without checking payment's state in your backend (capture can only succeed if `status` returned value is `successful`).
+
+P.S. It's not required for a successful app payments integration.
+
+:::
 
+</TabItem>
+</Tabs>
+</TabItem>
 <TabItem value="ios">
 
+## Prerequisites and Assumptions
+
+This guide assumes:
+
+- You already have backend services integrated with [MobilePay APIs](/docs/guides/app-payments/how-it-works).
+- Your app handles a custom URL scheme. See [Apple documentation](https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app) for more details.
+- You are generally familiar with developing applications on iOS.
```

Comment:
This guide has only mentioned Android stuff so far, so why the mentions about Apple and iOS here suddenly?

Reply:
Hmm or probably it's because this is the beginning of an iOS section. I just don't see any header about this.

Reply:
the above `<TabItem value="ios">` separates this into a new tab.
<img width="1680" alt="Screenshot 2021-11-10 at 13 43 19" src="https://user-images.githubusercontent.com/54581781/141107133-3d029f68-6b09-419d-be4c-6466ceeea6d0.png">